ext/pdo_mysql/tests/pdo_mysql_inTransaction.phpt

```diff
@@ -0,0 +1,68 @@
+--TEST--
+MySQL PDO class inTransaction
+--SKIPIF--
+<?php
+if (!extension_loaded('pdo') || !extension_loaded('pdo_mysql')) die('skip not loaded');
+require __DIR__ . '/config.inc';
+require __DIR__ . '/../../../ext/pdo/tests/pdo_test.inc';
+PDOTest::skip();
+?>
+--FILE--
+<?php
+require __DIR__ . '/../../../ext/pdo/tests/pdo_test.inc';
+
+const BEGIN = ['BEGIN', 'START TRANSACTION'];
+const END = ['COMMIT', 'ROLLBACK'];
+
+$db = PDOTest::test_factory(__DIR__ . '/common.phpt');
+// $pdo->setAttribute(PDO::ATTR_EMULATE_PREPARES, false); // mysql does not support
+for ($b = 0; $b < count(BEGIN); $b++) {
+    for ($e = 0; $e < count(END); $e++) {
+        foreach (['exec', 'query', 'execute'] as $w) {
+            foreach ([BEGIN[$b], END[$e]] as $command) {
+                switch ($w) {
+                    case 'exec':
+                        $db->exec($command);
+                        break;
+                    case'query':
+                        $db->query($command)->execute();
+                        break;
+                    case 'execute':
+                        /* EMULATE_PREPARES = QUERY */
+                        $db->prepare($command)->execute();
+                        break;
+                    default:
+                        assert(0);
+                }
+                var_dump($db->inTransaction());
```

Comment:
does consequent calls hit the server again and again?

Reply:
> does consequent calls hit the server again and again?

Do you mean that whether this call requires network IO? The MySQL server will tell the client if it is in a transaction in every response by a flag, so we can get the latest status no matter what command is executed

Reply:
> Do you mean that whether this call requires network IO?

yes

> The MySQL server will tell the client if it is in a transaction in every response by a flag, so we can get the latest status no matter what command is executed

so does this query this local flag or did it require network IO every time?

Reply:
not required, before sending a new request, the state of the transaction will not change, so the state we got by the last response is reliable

Reply:
@twose Thanks, I already hit this several times, your PR looks very good, 👍